### PR TITLE
feat:  Improving incremental development

### PR DESCRIPTION
**Pull request by mannaandpoem**
**Features**
-  Improving the functionality of incremental development in dev branch.
    
**Feature Docs**

**Influence**
- This feature will significantly improve the efficiency of our team's incremental development process.

**Result**
image
[log.txt](https://github.com/geekan/MetaGPT/files/14032169/log.txt)

**Other**

### CODE REVIEW

File: metagpt/actions/write_code.py [hunk 1]
```diff
@@ -101,22 +112,39 @@ async def run(self, *args, **kwargs) -> CodingContext:
...
+                code=code_context,
+                logs=logs,
+                feedback=bug_feedback.content if bug_feedback else "",
+                filename=self.i_context.filename,
+                summary_log=summary_doc.content if summary_doc else "",
+            )
+        else:
+            prompt = PROMPT_TEMPLATE.format(
+                design=coding_context.design_doc.content if coding_context.design_doc else "",
+                task=coding_context.task_doc.content if coding_context.task_doc else "",
```
Comment (iorisa) [inline metagpt/actions/write_code.py:141]: `tasks`? The place holder named `task` doesn't exists.

File: metagpt/actions/write_code.py [hunk 2]
```diff
@@ -127,20 +155,71 @@ async def run(self, *args, **kwargs) -> CodingContext:
...
         src_file_repo = project_repo.srcs
-        for filename in code_filenames:
-            if filename == exclude:
-                continue
-            doc = await src_file_repo.get(filename=filename)
-            if not doc:
-                continue
-            codes.append(f"----- {filename}\n" + doc.content)
+
+        if mode == "incremental":
```
Comment (iorisa) [inline metagpt/actions/write_code.py:189]: Define these literal key words in `const.py`

File: metagpt/actions/write_code_review.py [hunk 3]
```diff
@@ -136,22 +137,41 @@ async def write_code_review_and_rewrite(self, context_prompt, cr_prompt, filenam
 
     async def run(self, *args, **kwargs) -> CodingContext:
         iterative_code = self.i_context.code_doc.content
-        k = self.context.config.code_review_k_times or 1
+        # k = self.context.config.code_review_k_times or 1
```
Comment (iorisa) [inline metagpt/actions/write_code_review.py:140]: Are you sure you want to delete this line?

Comment (mannaandpoem) [reply metagpt/actions/write_code_review.py:140]: Sorry,  I forgot to change it back after testing

File: metagpt/actions/write_code.py (hunk 1 above)
Comment (mannaandpoem) [reply metagpt/actions/write_code.py:141]: Tasks have been modified to task in dev
image

File: metagpt/actions/write_code_review.py (hunk 3 above)
Comment (mannaandpoem) [reply metagpt/actions/write_code_review.py:140]: Perhaps A should be set to be passed in

File: metagpt/actions/write_code.py (hunk 2 above)
Comment (mannaandpoem) [reply metagpt/actions/write_code.py]: remove parameter mode in get_codes function. Use self.config.inc to judge condition

File: metagpt/actions/write_code.py (hunk 1 above)
Comment (iorisa) [reply metagpt/actions/write_code.py:140]: See the definition of `PROMPT_TEMPLATE`, it's named `tasks`
```
PROMPT_TEMPLATE = """
......
## Tasks
{tasks}
......
```

**Pull request merged**